src/librustc_resolve/lib.rs

```diff
@@ -3509,13 +3509,40 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
                                         format!("to call `{}::{}`", path_str, path_name),
                                 };
 
+                                let mut help_msg = String::new();
                                 if !msg.is_empty() {
                                     msg = format!(". Did you mean {}?", msg)
+                                } else {
+                                    // we check if this a module and if so, we display a help
+                                    // message
+                                    let name_path = path.segments.iter()
+                                                        .map(|seg| seg.identifier.name)
+                                                        .collect::<Vec<_>>();
+                                    let current_module = self.current_module.clone();
+
+                                    match self.resolve_module_path(current_module,
+                                                   &name_path[..],
+                                                   UseLexicalScope,
+                                                   expr.span,
+                                                   PathSearch) {
+                                        Success(_) => {
+                                            help_msg = format!("A module called `{module}` is \
+                                                                imported in the current scope. If \
+                                                                you want to call a function from \
+                                                                it, call it like this: \
+                                                                `{module}::some_function`.",
+                                                               module = &*path_name);
+                                        },
+                                        _ => {},
+                                    };
                                 }
 
                                 resolve_error(self,
                                               expr.span,
                                               ResolutionError::UnresolvedName(&*path_name, &*msg));
+                                if !help_msg.is_empty() {
```

Comment:
I don't mind, it's the same for me. I'll just provide an `Option` as extra parameter and we're good to go.